src/pages/profile.tsx

```diff
@@ -78,7 +113,95 @@ export default function Page() {
       <Input background="white" 
         value={newCity} onChange={e => setNewCity(e.target.value)} />
     </FormControl>           
+
+    <Divider margin={componentSpacing} />
+
+    <InterviewPreference data={pref.interviews} 
+      updateData={updateInterviewPref} /> 
+
     <Button onClick={handleSubmit} variant="brand">保存</Button>
     {isLoading && <Loader loadingText='保存中...'/>}
   </VStack>;
 }
+
+function InterviewPreference({ data, updateData } : {
+  data: InterviewPref;
+  updateData: (data: InterviewPref) => void;
+}) {
+  const [noMoreThan, setNoMoreThan] = useState(data?.limit?.noMoreThan || 0);
+  const [until, setUntil] = useState(data?.limit?.until ? 
+    new Date(data.limit.until) : oneMonthDate);
+  const [optIn, setOptIn] = useState(data?.optIn ?? undefined);
+
+  useEffect(() => {
+    setNoMoreThan(data?.limit?.noMoreThan ?? 0);
+    setUntil(new Date(data?.limit?.until ?? oneMonthDate));
+    setOptIn(data?.optIn ?? undefined);
+  }, [data]);
```

Comment:
dont use these intermediary state variables. directly use fields within `data`. See mentorProfile.tsx:121 and related lines for a reference implementation. I just added this file so you may want to rebase onto main.